packages/http/src/mocker/generator/JSONSchema.ts

```diff
@@ -98,6 +102,26 @@ export function sortSchemaAlphabetically(source: any): any {
   return source;
 }
 
-export function generateStatic(resource: IHttpOperation, source: JSONSchema): Either<Error, unknown> {
-  return tryCatch(() => sampler.sample(source, {}, resource), toError);
+export function generateStatic(operation: IHttpOperation, source: JSONSchema): Either<Error, unknown> {
+  return pipe(
+    tryCatch(() => sampler.sample(source, {}, operation), toError),
+    E.mapLeft((err: Error) => {
+      if (err instanceof sampler.SchemaSizeExceededError) {
+        return new SchemaTooComplexGeneratorError(operation, err);
+      }
+      return err;
+    })
+  );
+}
+
+export class GeneratorError extends Error {}
+
+export class SchemaTooComplexGeneratorError extends GeneratorError {
+  constructor(operation: IHttpOperation, public readonly cause: Error) {
+    super(
+      `The operation ${operation.method.toUpperCase()} ${
+        operation.path
+      } references a JSON Schema that is too complex to generate.`
```

Comment:
Is there a way to state precisely in what way/dimension the spec is too complex here?  As a Prism user who sees this, what can I do to work-around the problem or simplify the spec to make it fall within the boundary?

Reply:
we'd have to make json-schema-sampler and json-schema-faker to provide that context. the sampler code is all vanilla js so its pretty terse to follow

Reply:
Is this a case where we could add to the Prism docs we publish, and include a URL here for more info?  I know we haven't done this before, but I'm just trying to find _some_ way to give users information they can act on (or at least understand).  I just don't think, as a Prism user, I'd be able to understand "too complex to generate" enough to solve my own problem.